horace_core/sqw/file_io/@dnd_binfile_common/dnd_binfile_common.m

```diff
@@ -474,18 +499,32 @@ function check_obj_initated_properly(obj)
             obj.file_id_ = 0;
         end
         %
-        function obj = activate(obj)
-            % open respective file for reading without reading any
-            % supplementary file information. Assume that this information
-            % is correct.
+        function obj = activate(obj, read_or_write)
+            % Open respective file in read or write mode without reading any
+            % supplementary file information. Assume that header information
+            % stored on this object is correct.
             %
-            % To use for MPI transfers between workers when open file can
+            % Can be used for MPI transfers between workers when open file can
             % not be transferred between workers but everything else can
+            %
+            % Input
+            % -----
+            %
+            % read_or_write   Char array. If 'read' open the file in read-only
+            %                 mode If 'write' open the file in read/write mode.
+            %                 Default is 'read'.
+            %
+            if nargin == 1
+                permission = 'rb';
```

Comment:
you have `rb` in two places now; this should be a call to `get_fopen_permission_("read")` or adapt that function to handle the default case